### ISSUE TITLE
On import DAE files are ignoring created materials from script

### ISSUE THREAD

**Issue by LevyImage** [Markemp/Asset-Importer#3]
The script output is creating materials in Blender but, when DAE files begin their import toward the bottom of script, it seems the DAE files are essentially ignoring the material created by the script. They instead seem to  create additional materials which are included in the DAE files themselves instead of using the existing materials with matching names. Not sure if this behavior is correctable via blender python script window.

**Comment by Markemp** [comment]
Are you using converted .mtl files?  It can't read the binary versions in CIG.  I *think* you can get away with not having exact same .mtl file name (using holoxplorer to create deserialized, regular mtl versions), as it's just creating a bunch of materials that then get assigned by material name.

The extra materials created in Blender when you import a .dae file are due to Blender not clearing out Material references after a material is deleted.  Instead, the material continues to exist with no Users, until the file is saved, closed, and reloaded.  Upon reload, if Blender sees there are no Users for a material, it will be deleted.  (There is a way to save materials with no Users, but no point in this particular application.)  As a result, when you import a few dozen .dae files, you'll see material.000, material.001, etc.  There is a reason Blender does this, but I can't remember it off the top of my head.

Although, rereading your report... you're seeing the proper Materials being created, but that material doesn't get assigned to the Object in Blender?  Do you have a directory in SC I can test out?

**Comment by LevyImage** [comment]
Yes sir, materials are created from script but they aren't getting assigned to their respective objects after the dae files import. If you remove the LODs to save yourself some hassle (otherwise will make your outliner a mess), for both 2.6.3 and PTU 3.0 `\Data\Objects\Spaceships\Ships\ANVL\Hornet\F7CM`
